packages/safe-apps-sdk/src/types/permissions.ts

```diff
@@ -0,0 +1,36 @@
+export type Permission = {
+  parentCapability: string;
+  invoker: string;
+  date?: number;
+  // Part of the EIP-2255 spec. Added for future use.
+  // Useful for representing specific restrictions applied to the permitted method
+  // Will require a concrete implementation when used
+  // See: https://eips.ethereum.org/EIPS/eip-2255
+  caveats?: PermissionCaveat[];
+};
+
+export type PermissionRequest = {
+  [method: string]: Record<string, unknown>;
+};
+
+export type PermissionCaveat = {
+  type: string;
+  value?: unknown;
+  name?: string;
+};
+
+export const PERMISSIONS_REQUEST_REJECTED = 4001;
+
+export class PermissionsError extends Error {
+  public code: number;
+  public data?: unknown;
+
+  constructor(message: string, code: number, data?: unknown) {
+    super(message);
+
+    this.code = code;
+    this.data = data;
+
+    Object.setPrototypeOf(this, PermissionsError.prototype);
```

Comment:
Why is this line needed?

Reply:
That's because this https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
Could you please include this as a comment to explain this line?

Reply:
of course !! done